xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-ui/src/main/resources/Panels/PanelSheet.xml

```diff
@@ -39,14 +39,14 @@
   <content>{{velocity output="false"}}
 #macro(displayPanelProperty $obj $propName)
   ; &lt;label#if ($xcontext.action == 'edit' || $xcontext.action == 'inline') for="${class.getName()}_${obj.number}_${propName}"#end&gt;$services.localization.render("${class.getName()}_${propName}")&lt;/label&gt;
-  : $doc.display($propName, $obj)
+  : $tdoc.display($propName, $obj)
 #end
 
-#set ($paneldoc = $doc)
+#set ($paneldoc = $tdoc)
 {{/velocity}}
 
 {{velocity}}
-#set ($obj = $doc.getObject('Panels.PanelClass'))
+#set ($obj = $tdoc.getObject('Panels.PanelClass'))
```

Comment:
You sure about that ? Translation documents are not supposed to contain xobjects.

Reply:
it's actually what makes it working :) we are always using the tdoc in the SaveAction. So if you are using an edit with `language=something` in a multilingual wiki, the translation will contain an xobject.